### PR TITLE
all: remove discussions backend code and config

### PR DESCRIPTION
**Pull request by keegancsmith**
See individual commits. Still need to drop the tables, but that managed to do this in what I timeboxed. We may want to backup the tables on sourcegraph.com before dropping though so better to merge this first.

Fixes https://github.com/sourcegraph/sourcegraph/issues/9649

### CODE REVIEW

File: schema/site.schema.json [hunk 1]
```diff
@@ -497,7 +491,7 @@
     },
     "email.imap": {
       "title": "IMAPServerConfig",
-      "description": "Optional. The IMAP server used to retrieve emails (such as code discussion reply emails).",
+      "description": "DEPRECATED. Will be removed in 3.17. Was used by the now removed discussions feature.",
```
Comment (felixfbecker) [inline schema/site.schema.json:494]: Do saved search email notifications use a different config?

**Comment by felixfbecker** [comment]
I'm surprised there is 0% coverage change from such a large code removal. Is something wrong with Go coverage tracking?

File: schema/site.schema.json (hunk 1 above)
Comment (keegancsmith) [reply schema/site.schema.json:494]: imap is for reading email. the smtp config is for sending and remains.

**Comment by keegancsmith** [comment]
> I'm surprised there is 0% coverage change from such a large code removal. Is something wrong with Go coverage tracking?

codecov has partial availability due to being rate limited by GitHub. https://status.codecov.io/ Not sure how well the "patch" reporting works with code removal. But looking at the UI it is recording coverage for Go.

**Comment by keegancsmith** [comment]
diff with normal 'cover' command. coverage went up in DB, and down slightly in app/ui. Removed a bunch of packages with varying code coverage.

```
4c4
< github.com/sourcegraph/sourcegraph/cmd/frontend/db 52.1%
---
> github.com/sourcegraph/sourcegraph/cmd/frontend/db 54.9%
15c15
< github.com/sourcegraph/sourcegraph/cmd/frontend/internal/app/ui 41.6%
---
> github.com/sourcegraph/sourcegraph/cmd/frontend/internal/app/ui 41.4%
25,29d24
< github.com/sourcegraph/sourcegraph/cmd/frontend/internal/pkg/discussions 7.3%
< github.com/sourcegraph/sourcegraph/cmd/frontend/internal/pkg/discussions/mailreply 12.7%
< github.com/sourcegraph/sourcegraph/cmd/frontend/internal/pkg/discussions/mentions 100.0%
< github.com/sourcegraph/sourcegraph/cmd/frontend/internal/pkg/discussions/ratelimit 29.7%
< github.com/sourcegraph/sourcegraph/cmd/frontend/internal/pkg/discussions/searchquery 100.0%
```

Commands I ran for future people who are interested in this.
```shell
go -cover ./... | awk '/^ok/ {print $2 " " $5}' | sort > after.txt
git checkout beforecommit
go -cover ./... | awk '/^ok/ {print $2 " " $5}' | sort > before.txt
diff before.txt after.txt
```

**Pull request merged**